sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts

```diff
@@ -191,22 +196,56 @@ export class MSFAANumberService extends RecordDataModelService<MSFAANumber> {
     cancelledDate: Date,
     newIssuingProvince: string,
   ): Promise<UpdateResult> {
-    if (!cancelledDate || !newIssuingProvince) {
-      throw new Error(
-        "Not all required fields to update a received MSFAA record were provided.",
+    return this.dataSource.transaction(async () => {
+      // Keeping both MSFAA Cancellation update and saving of MSFAA Notification message as a part of the same transaction.
+      if (!cancelledDate || !newIssuingProvince) {
+        throw new Error(
+          "Not all required fields to update a received MSFAA record were provided.",
+        );
+      }
+      // Update the MSFAA record that needs to be cancelled.
+      const updateResult = await this.repo.update(
+        {
+          msfaaNumber: msfaaNumber,
+          cancelledDate: IsNull(),
+          newIssuingProvince: IsNull(),
+        },
+        {
+          cancelledDate: getISODateOnlyString(cancelledDate),
+          newIssuingProvince,
+        },
       );
-    }
-
-    return this.repo.update(
-      {
-        msfaaNumber: msfaaNumber,
-        cancelledDate: IsNull(),
-        newIssuingProvince: IsNull(),
-      },
-      {
-        cancelledDate: getISODateOnlyString(cancelledDate),
-        newIssuingProvince,
-      },
-    );
+      // Get the student associated with the cancelled MSFAA record.
+      const msfaaRecord = await this.repo.findOne({
```

Comment:
Please move the `findOne` inside the `if (updateResult.affected)` since it is only needed if the notification will be sent.